crates/cext/src/circuit.rs

```diff
@@ -682,3 +683,112 @@ pub unsafe extern "C" fn qk_circuit_delay(
 
     ExitCode::Success
 }
+
+/// @ingroup QkCircuit
+/// Compose a circuit into other circuit
+///
+/// @param target A pointer to the circuit to compose a circuit
+/// @param other A pointer to the circuit to be composed into
+/// @param qubits A pointer to array of qubits used to map other circuit to target
+/// @param clbits A pointer to array of clbits used to map other circuit to target
```

Comment:
We should include a bit more detail on this, including how the mapping is done, the fact that the lengths of the arrays are inferred from `other.num_qubits`, etc.